samples/dotnet/EdgeX509AuthDownstreamDevice/Program.cs

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Pkcs;
+using Org.BouncyCastle.Security;
+
+
+using Microsoft.Azure.Devices.Client;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    class Program
+    {
+
+        // 1) Obtain the IoT Hub hostname and device id for your downstream device.
+        //      Update the IOTHUB_HOSTNAME and DEVICE_ID in the Properties/launchSettings.json file.
+        // 2) Obtain the Edge device hostname: The edge device hostname is the hostname set in
+        //    the config.yaml of the Edge device to which this sample will connect to.
+        //      Update the IOTEDGE_GATEWAY_HOSTNAME in the Properties/launchSettings.json file.
+        // 3) Obtain the trusted CA certificate required to trust the Edge gateway.
+        // In the docs this would be the azure-iot-test-only.root.ca.cert.pem 
+        private static readonly string iothubHostname = Environment.GetEnvironmentVariable("IOTHUB_HOSTNAME");
+        private static readonly string downstreamDeviceId = Environment.GetEnvironmentVariable("DEVICE_ID");
+        private static readonly string iotEdgeGatewayHostname = Environment.GetEnvironmentVariable("IOTEDGE_GATEWAY_HOSTNAME");
+        private static readonly string deviceIdentityCertPath = Environment.GetEnvironmentVariable("DEVICE_IDENTITY_X509_CERTIFICATE_PEM_PATH");
+        private static readonly string deviceIdentityPrivateKeyPath = Environment.GetEnvironmentVariable("DEVICE_IDENTITY_X509_CERTIFICATE_KEY_PEM_PATH");
+        private static readonly string trustedCACertPath = Environment.GetEnvironmentVariable("IOTEDGE_TRUSTED_CA_CERTIFICATE_PEM_PATH");
+        private static int MESSAGE_COUNT = 10;
+        private const int TEMPERATURE_THRESHOLD = 30;
+
+        public static IEnumerable<X509Certificate2> GetCertificatesFromPem(IEnumerable<string> rawPemCerts) =>
+            rawPemCerts
+                .Select(c => System.Text.Encoding.UTF8.GetBytes(c))
+                .Select(c => new X509Certificate2(c))
+                .ToList();
+
+        public static IList<string> ParsePemCerts(string pemCerts)
```

Comment:
CertificateHelper from Util has parsing of the pem file, can we use that?

Reply:
The reason I wrote this way was to make sure someone could use this sample as a standalone without the need for the Edge project.

Reply:
Resolving for now. LMK if you have any concerns.